### PR TITLE
Fix/role marking bugs

### LINKED ISSUE #237: Inappropriate permissions for direct links to assignments
If a teacher for some course does not have the global permission to add courses they cannot acces direct links to assignment (i.e. in the upcoming deadlines section). Instead they get a 404.

The permissions for these direct links should be checked for the specific course they link to, rather than this global (irrelevant!) permission to add courses.

### LINKED ISSUE #244: Unpublished points are included in total given in progress nodes
(For students)

Also shows negative points required, should maybe show a different message / have some visual in the EDAG showing the user they have hit the required points (might already exist but not calculated from incorrect total).

Additionally an enhancement could be made on the teacher side to show a different message that shows a distinction between published/unpublished points. Could also apply to various progress bars using different colors.

### PR DESCRIPTION
**Pull request by engelhamer**
- Fixed #237 
- Fixed #244 
- Moved journal redirection to assignment page for correct and simple behaviour of shared assignments between courses.

### CODE REVIEW

**Comment by engelhamer** [comment]
@Airgetfrog such as? I don't see any comments from you.

File: src/vue/src/components/assets/BreadCrumb.vue [hunk 1]
```diff
@@ -124,7 +124,7 @@ export default {
 
             if ((pageName === 'Home' && this.$root.isAdmin()) ||
                (pageName === 'Course' && this.$root.canEditCourse()) ||
-               (pageName === 'Assignment' && this.$root.canEditCourse())) {
+               (pageName === 'Assignment' && this.$root.canEditAssignment())) {
```
Comment (Airgetfrog) [inline src/vue/src/components/assets/BreadCrumb.vue]: Edit and change format buttons in the assignment editor should have their permissions changed accordingly.

File: src/vue/src/components/journal/JournalNonStudent.vue [hunk 2]
```diff
@@ -13,10 +13,12 @@
...
                     <div v-else-if="nodes[currentNode].type == 'p'">
-                        <b-card class="no-hover" :class="'pink-border'">
-                            <h2 class="mb-2">Progress: {{nodes[currentNode].target}} points</h2>
-                            {{progressNodes[nodes[currentNode].nID]}} out of {{nodes[currentNode].target}} points.<br/>
-                            {{nodes[currentNode].target - progressNodes[nodes[currentNode].nID]}} more required before {{$root.beautifyDate(nodes[currentNode].deadline)}}.
+                        <b-card class="no-hover" :class="getProgressBorderClass()">
+                            <h2 class="mb-2">Progress: {{ nodes[currentNode].target }} points</h2>
+                            <span v-if="progressPointsLeft > 0">
+                                <b>{{ progressNodes[nodes[currentNode].nID] }}</b> out of <b>{{ nodes[currentNode].target }}</b> points.<br/>
+                                <b>{{ progressPointsLeft }}</b> more required before <b>{{ $root.beautifyDeadline(nodes[currentNode].deadline) }}</b>.
```
Comment (Airgetfrog) [inline src/vue/src/components/journal/JournalNonStudent.vue]: Should this not be beautifyDate?

File: src/vue/src/components/journal/JournalStudent.vue [hunk 3]
```diff
@@ -32,10 +32,12 @@
...
                     <div v-else-if="nodes[currentNode].type == 'p'">
-                        <b-card class="no-hover" :class="'pink-border'">
-                            <h2 class="mb-2">Progress: {{nodes[currentNode].target}} points</h2>
-                            {{ progressNodes[nodes[currentNode].nID] }} out of {{ nodes[currentNode].target }} points.<br/>
-                            {{ nodes[currentNode].target - progressNodes[nodes[currentNode].nID] }} more required before {{ $root.beautifyDate(nodes[currentNode].deadline) }}.
+                        <b-card class="no-hover" :class="getProgressBorderClass()">
+                            <h2 class="mb-2">Progress: {{ nodes[currentNode].target }} points</h2>
+                            <span v-if="progressPointsLeft > 0">
+                                <b>{{ progressNodes[nodes[currentNode].nID] }}</b> out of <b>{{ nodes[currentNode].target }}</b> points.<br/>
+                                <b>{{ progressPointsLeft }}</b> more required before <b>{{ $root.beautifyDeadline(nodes[currentNode].deadline) }}</b>.
```
Comment (Airgetfrog) [inline src/vue/src/components/journal/JournalStudent.vue]: Should this not be beautifyDate?

**Comment by Airgetfrog** [comment]
how strange. ive added them back in now

**Pull request merged**